File: app-kotlin/android/app/src/main/java/com/pedidosexpress/BusinessTypeHelper.kt
package com.pedidosexpress

/**
 * Helper para obter labels dinâmicos baseados no tipo de negócio
 */
object BusinessTypeHelper {
    private fun getLabel(user: User?, defaultLabel: String, dentistaLabel: String): String {
        return if (user?.isDentista == true) dentistaLabel else defaultLabel
    }
    
    // Labels principais
    fun ordersLabel(user: User?): String {
        return getLabel(user, "Pedidos", "Agendamentos")
    }
    
    fun menuLabel(user: User?): String {
        return getLabel(user, "Cardápio", "Procedimentos")
    }
    
    fun orderLabel(user: User?): String {
        return getLabel(user, "Pedido", "Agendamento")
    }
    
    fun ordersTodayLabel(user: User?): String {
        return getLabel(user, "Pedidos Hoje", "Agendamentos Hoje")
    }
    
    fun itemsLabel(user: User?): String {
        return getLabel(user, "Itens", "Procedimentos")
    }
    
    fun itemLabel(user: User?): String {
        return getLabel(user, "Item", "Procedimento")
    }
    
    // Labels para segmentos
    fun pendingOrdersLabel(user: User?): String {
        return getLabel(user, "Pedidos", "Agendamentos")
    }
    
    fun outForDeliveryLabel(user: User?): String {
        return getLabel(user, "Rota", "Confirmados")
    }
    
    fun finishedOrdersLabel(user: User?): String {
        return getLabel(user, "Entregues", "Concluídos")
    }
}

File: app-swift/PedidosExpress/Helpers/BusinessTypeHelper.swift
import Foundation

/**
 * Helper para obter labels dinâmicos baseados no tipo de negócio.
 * Considera AuthService.getBusinessType() quando o User não traz business_type (ex.: backend não envia).
 */
class BusinessTypeHelper {
    private static var effectiveIsBarber: Bool {
        AuthService().getBusinessType()?.uppercased() == "BARBEIRO"
    }

    private static var effectiveIsDentista: Bool {
        AuthService().getBusinessType()?.uppercased() == "DENTISTA"
    }

    static func getLabel(for user: User?, defaultLabel: String, dentistaLabel: String) -> String {
        let useAgendaLabel = (user?.isDentista ?? effectiveIsDentista) || (user?.isBarbeiro ?? effectiveIsBarber)
        return useAgendaLabel ? dentistaLabel : defaultLabel
    }

    static func ordersLabel(for user: User?) -> String {
        return getLabel(for: user, defaultLabel: "Pedidos", dentistaLabel: "Agendamentos")
    }

    static func agendaTabLabel(for user: User?) -> String {
        if user?.isBarbeiro == true || effectiveIsBarber { return "Agenda" }
        if user?.isDentista == true || effectiveIsDentista { return "Agendamentos" }
        return "Pedidos"
    }

    static func supportTabLabel(for user: User?) -> String {
        return (user?.isBarbeiro == true || effectiveIsBarber) ? "Atendimento" : "Suporte"
    }

    static func menuLabel(for user: User?) -> String {
        if user?.isBarbeiro == true || effectiveIsBarber { return "Serviços" }
        if user?.isDentista == true || effectiveIsDentista { return "Procedimentos" }
        return "Cardápio"
    }
    
    static func orderLabel(for user: User?) -> String {
        return getLabel(for: user, defaultLabel: "Pedido", dentistaLabel: "Agendamento")
    }
    
    static func ordersTodayLabel(for user: User?) -> String {
        return getLabel(for: user, defaultLabel: "Pedidos Hoje", dentistaLabel: "Agendamentos Hoje")
    }
    
    static func itemsLabel(for user: User?) -> String {
        return getLabel(for: user, defaultLabel: "Itens", dentistaLabel: "Procedimentos")
    }
    
    static func itemLabel(for user: User?) -> String {
        return getLabel(for: user, defaultLabel: "Item", dentistaLabel: "Procedimento")
    }
    
    // Labels para segmentos
    static func pendingOrdersLabel(for user: User?) -> String {
        return getLabel(for: user, defaultLabel: "Pedidos", dentistaLabel: "Agendamentos")
    }
    
    static func outForDeliveryLabel(for user: User?) -> String {
        return getLabel(for: user, defaultLabel: "Rota", dentistaLabel: "Confirmados")
    }
    
    static func finishedOrdersLabel(for user: User?) -> String {
        return getLabel(for: user, defaultLabel: "Entregues", dentistaLabel: "Concluídos")
    }
}
